RHEL/6/input/checks/ftp_disable_uploads.xml

```diff
@@ -0,0 +1,26 @@
+<def-group>
+  <definition class="compliance" id="ftp_disable_uploads" version="1">
+    <metadata>
+      <title>Disable FTP Uploads</title>
+      <affected family="unix">
+        <platform>Red Hat Enterprise Linux 6</platform>
+      </affected>
+      <description>Disable FTP Uploads.</description>
+      <reference source="galford" ref_id="201516" ref_url="test_attestation" />
+    </metadata>
+    <criteria operator="OR">
+      <extend_definition comment="vsftpd package is not installed" negate="true" definition_ref="package_vsftpd_installed" />
+      <criterion comment="Disable uploads" test_ref="test_ftp_disable_uploads" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="Disable uploads" id="test_ftp_disable_uploads" version="1">
+    <ind:object object_ref="object_ftp_disable_uploads" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object comment="Disable uploads" id="object_ftp_disable_uploads" version="1">
+    <ind:filepath>/etc/vsftpd/vsftpd.conf</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*write_enable[\s]*=[\s]*NO$</ind:pattern>
```

Comment:
```vsftpd``` doesn't seem to allow whitespace around config key and config value assignment (around '=' character).
Leading space (' ' or '\t' is OK):
```
# cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 6.6 (Santiago)
# rpm -q vsftpd
vsftpd-2.2.2-13.el6_6.1.x86_64
# cat -A /etc/vsftpd/vsftpd.conf | grep write_enable
^Iwrite_enable=NO$
#anon_mkdir_write_enable=YES$
# service vsftpd restart
Shutting down vsftpd:                                      [  OK  ]
Starting vsftpd for vsftpd:                                [  OK  ]
```
Same for leading ' ' / space characters (it's OK):
```
# cat -A /etc/vsftpd/vsftpd.conf | grep write_enable
    write_enable=NO$
#anon_mkdir_write_enable=YES$
# service vsftpd restart
Shutting down vsftpd:                                      [  OK  ]
Starting vsftpd for vsftpd:                                [  OK  ]
```
But whitespace (' ' or '\t') isn't allowed around '=' character:
```
# cat -A /etc/vsftpd/vsftpd.conf | grep write_enable
write_enable^I=NO$
#anon_mkdir_write_enable=YES$
# service vsftpd restart
Shutting down vsftpd:                                      [  OK  ]
Starting vsftpd for vsftpd: 500 OOPS: unrecognised variable in config file: write_enable	
                                                           [FAILED]
```
Same for space character:
```
# cat -A /etc/vsftpd/vsftpd.conf | grep write_enable
write_enable=    NO$
#anon_mkdir_write_enable=YES$
# service vsftpd restart
Shutting down vsftpd:                                      [FAILED]
Starting vsftpd for vsftpd: 500 OOPS: bad bool value in config file for: write_enable
                                                           [FAILED]
```
=> the corresponding pattern should be updated not to allow [\s]* around equals '=' character.